### ISSUE TITLE
Timeslot length should be selectable by the user

### ISSUE THREAD

**Issue by laander** [timekit-io/booking-js#61]

**Comment by alialkhatib** [comment]
I'm making this work by implementing a really hacky form that passes a time duration to the timekit filter. Passing the selection (or slider's value or whatever) to the configuration and reloading the calendar should allow the user to dictate the duration.

**Comment by laander** [comment]
@alialkhatib Thanks for chiming in. Yeah, this is what we recommend too for the time being. As you're saying, the `length` parameter in Find Time can take any duration and as the booking.js config is all JS, you can simply pass it dynamically upon instantiation.

In the near future, I'm considering building a "pick a duration" dropdown into the widget, where visitors can pick from some predefined lengths. We're pretty short on UI space though, but shouldn't be that hard to implement.

**Comment by cyntss** [comment]
I agree. ATM i find myself implementing a terrible workaround to solve this. In my use case, the web site is for booking day care (8hs), dog sitting (24hs) or dog walking (1hs).
The user should be able to select the time slot starting from any point, plus, in the dog sitting case, the user should be able to pick as many days as wanted. I see a use case here as well for the other, where a user want to select 5 mondays in a row for dog walking at 10am. 

I think this will be very useful for other use cases such: house rental, car rental, etc.

Now.. I just see the message from @laander and I will try to implement the range selector myself...although  I have the feeling the user just wants to click around in the calendar because it looks simpler.
